lib/cylc/review.py

```diff
@@ -912,6 +920,45 @@ def _check_string_for_path(string):
         if os.path.split(string)[0] != '':
             raise cherrypy.HTTPError(403)
 
+    @classmethod
+    def _check_link_path(cls, path, suite):
+        """Raise HTTP 403 error if the path is not under cylc-run/<suite>.
+
+        The files "cylc review" is intended to serve may be symlinked to
+        other parts of the filesystem, however, the linked files should
+        always reside in directories managed by Cylc which should always
+        sit under the path "cylc-run/<suite>".
+
+        Examples:
+            # OK
+            >>> CylcReviewService._check_link_path(
+            ...     os.path.expanduser('~/cylc-run/elephant'), 'elephant'
+            ... )
+
+            # BAD
+            >>> CylcReviewService._check_link_path(
+            ...     os.path.expanduser('~/cylc-run/elephant'), 'shrew'
```

Comment:
As a member of the Afrotherium clade elephant shrews are more closely related to elephants than shrews.

Reply:
TIL